v2/cmd/controller/main.go

```diff
@@ -179,6 +179,13 @@ func getDefaultAzureCredential(cfg config.Values, setupLog logr.Logger) (azcore.
 		if err != nil {
 			return nil, errors.Wrapf(err, "unable to get workload identity credential")
 		}
+	} else if cert := os.Getenv(config.ClientCertificateVar); cert != "" {
+		certPassword := os.Getenv(config.ClientCertificatePasswordVar)
+		credential, err = identity.NewClientCertificateCredential(cfg.TenantID, cfg.ClientID, []byte(cert), []byte(certPassword))
+		if err != nil {
+			setupLog.Error(err, "unable to get client certificate credential")
+			os.Exit(1)
```

Comment:
This should return an error, not `os.Exit`.

Let the caller decide to exit or not, like with the other errors here.